stream/partitioner.go

```diff
@@ -0,0 +1,92 @@
+package stream
+
+import (
+	"fmt"
+	"hash"
+	"hash/fnv"
+	"sync"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/pkg/errors"
+)
+
+// HashPartitioner is a wrapper struct around the *kafka.Producer from
+// confluent-kafka-go/kafka. It intercepts the Produce method and overrides the
+// TopicPartition.Partition based on the hash of the message.Key.
+type HashPartitioner struct {
```

Comment:
I'm wondering if we should keep using sarama's partitioning scheme. An option would be to manually mark all the messages we have in `outbound_event_queue` as `processed = false`, clear out the topics and reprocess al the messages using the new scheme.

It would mean that we'd re-import everywhere that we listen to pg2kafka, but those results should stay unchanged.

Reply:
We _do_ need to partition by primary key right? Otherwise we can't promise in-order processing.

Reply:
yes but like I mentioned with my PR the default is `consistent_random` see https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md

basically it mean key are partitioned using crc hash and null keys are distributed randomly

Reply:
Yes, librdkafka uses the same partitioning scheme that is used in the java implementation, it doesn't really matter which version you use, but it would save us a bunch of code to maintain if we could just use the default.